classes/user_extractor.php

```diff
@@ -76,7 +76,26 @@ public static function get_user(string $fieldname, string $fieldvalue, bool $ins
             // Check if requested field exists, required for Totara compatibility.
             $fields = array_merge(\core_user::AUTHSYNCFIELDS, ['id', 'username']);
             if (in_array($fieldname, $fields)) {
-                $user = get_complete_user_data($fieldname, $fieldvalue);
+                if ($insensitive) {
+                    $fieldsql = " AND ". $DB->sql_equal($fieldname, ':fieldvalue', !$insensitive);
+                    $params['fieldname'] = $fieldname;
+                    $params['fieldvalue'] = $fieldvalue;
+                    $params['mnethostid'] = $CFG->mnet_localhost_id;
+
+                    $sql = "SELECT id
```

Comment:
If we use a custom SQL in some cases, would it be better to use it all the time? Would probably require a little bit of refactoring to remove a duplicating code.